machine_test.go

```diff
@@ -1772,6 +1821,327 @@ func TestCreateSnapshot(t *testing.T) {
+			createSnapshot: func(ctx context.Context, machineLogger *logrus.Logger, socketPath, memPath, snapPath string) {
+
+			},
+
+			loadSnapshot: func(ctx context.Context, machineLogger *logrus.Logger, socketPath, memPath, snapPath string) {
+				cfg := createValidConfig(t, socketPath+".load")
+				m, err := NewMachine(ctx, cfg, func(m *Machine) {
+					// Rewriting m.cmd partially wouldn't work since Cmd has
+					// some unexported members
+					args := m.cmd.Args[1:]
+					m.cmd = exec.Command(getFirecrackerBinaryPath(), args...)
+				}, WithLogger(logrus.NewEntry(machineLogger)))
+				require.NoError(t, err)
+
+				err = m.Start(ctx, WithSnapshot(memPath, snapPath))
+				require.Error(t, err)
+			},
+		},
+		{
+			name: "TestLoadSnapshot and check contents (via ssh)",
+			createSnapshot: func(ctx context.Context, machineLogger *logrus.Logger, socketPath, memPath, snapPath string) {
+				cniConfPath := fmt.Sprintf("%s/%s.conflist", cniConfDir, networkName)
+				err := writeCNIConfWithHostLocalSubnet(cniConfPath, networkName, subnet)
+				require.NoError(t, err)
+				defer os.Remove(cniConfPath)
+
+				networkInterface := NetworkInterface{
+					CNIConfiguration: &CNIConfiguration{
+						NetworkName: networkName,
+						IfName:      ifName,
+						ConfDir:     cniConfDir,
+						BinPath:     cniBinPath,
+						VMIfName:    "eth0",
+					},
+				}
+				cfg := createValidConfig(t, fmt.Sprintf("%s.create", socketPath),
+					withRootDrive(rootfsPath),
+					withNetworkInterface(networkInterface),
+				)
+
+				cmd := VMCommandBuilder{}.WithSocketPath(fmt.Sprintf("%s.create", socketPath)).WithBin(getFirecrackerBinaryPath()).Build(ctx)
+
+				m, err := NewMachine(ctx, cfg, WithProcessRunner(cmd))
+				require.NoError(t, err)
+
+				err = m.Start(ctx)
+				require.NoError(t, err)
+				defer m.StopVMM()
+				defer func() {
+					if err := m.Shutdown(ctx); err != nil {
+						t.Logf("error on vm shutdown: %v", err)
+					}
+				}()
+
+				var client *ssh.Client
+				for i := 0; i < maxRetries; i++ {
+					client, err = connectToVM(m, sshKeyPath)
+					if err != nil {
+						time.Sleep(backoffTimeMs * time.Millisecond)
+					} else {
+						break
+					}
+				}
+				require.NoError(t, err)
+				defer client.Close()
+
+				ipToRestore = m.Cfg.NetworkInterfaces.staticIPInterface().StaticConfiguration.IPConfiguration.IPAddr.IP.String()
+
+				session, err := client.NewSession()
+				require.NoError(t, err)
+				defer session.Close()
+
+				err = session.Start(`sleep 422`) // Should be asynchronous
+				require.NoError(t, err)
+
+				err = m.PauseVM(ctx)
+				require.NoError(t, err)
+
+				err = m.CreateSnapshot(ctx, memPath, snapPath)
+				require.NoError(t, err)
+			},
+
+			loadSnapshot: func(ctx context.Context, machineLogger *logrus.Logger, socketPath, memPath, snapPath string) {
+				var ipFreed bool = false
+				var err error
+
+				for i := 0; i < maxRetries; i++ {
+					if _, err = os.Stat("/var/lib/cni/networks/fcnet/" + ipToRestore); err == nil {
```

Comment:
1. Let's create a method variable `const cniCacheDir = "/var/lib/cni"` around line 1898 with the network name. 
2. Also let's add a comment along the lines of `// Using default cache directory to ensure collision avoidance on IP allocations`
3. Finally here it will look like `os.Stat(fmt.Sprintf("%s/networks/%s/%s", cniCacheDir, networkName, ipToRestore))`